### PR TITLE
finish the doxygen for g2, issue #55

### PR DESCRIPTION
**Pull request by Hang-Lei-NOAA**
This include the full doxygen added into the g2 library source codes.
Agreed by Ed, I pull them all in this PR to accelerate the process.
I have addressed all warnings in the testing.
It is included in
https://hang-lei-noaa.github.io/NCEPLIBS/NCEPLIBS-g2/html/files.html

### CODE REVIEW

**Review by edwardhartnett — APPROVED** [review]

File: src/putgb2.f [hunk 1]
```diff
@@ -1,193 +1,172 @@
+C>    @file
+C>    @brief This subroutine read and unpack sections 6 and 7 from ah
+C>    grib2 message.
+C>    @author Stephen Gilbert @date 2002-01-11
+C>
+
+C>    This subroutine packs a single field into a grib2 message. and
```
Comment (GeorgeGayno-NOAA) [inline src/putgb2.f]: Eliminate period.

File: src/putgb2.f [hunk 2]
```diff
@@ -1,193 +1,172 @@
...
+C>    note that file/unit lugb should be opened woth a call to
+C>    subroutine baopenw before this routine is called.
+C>    The information to be packed into the grib field is stored in a
+C>    derived type variable, gfld. gfld is of type gribfield, which is
+C>    defined in module grib_mod, so users of this routine will need to
+C>    include the line "use grib_mod" in their calling routine. Each
+C>    component of the gribfield type is described in the input argument
+C>    list section below.
+C>    
+C>    PROGRAM HISTORY LOG:
```
Comment (GeorgeGayno-NOAA) [inline src/putgb2.f:18]: Make this a bullet list.

Comment (Hang-Lei-NOAA) [reply src/putgb2.f:18]: Thanks, it has been corrected.

**Review by GeorgeGayno-NOAA — APPROVED** [review]
Looks good.

**Comment by GeorgeGayno-NOAA** [comment]
@Hang-Lei-NOAA The only remaining tasks are to eliminate these warnings:

```
/gpfs/dell2/emc/modeling/noscrub/George.Gayno/g2_lib.git/NCEPLIBS-g2-1/src/enc_jpeg2000.c:77: warning: Member MAXOPTSSIZE (macro definition) of file enc_jpeg2000.c is not documented.

/gpfs/dell2/emc/modeling/noscrub/George.Gayno/g2_lib.git/NCEPLIBS-g2-1/src/drstemplates.f:1: warning: Member maxlen (variable) of class drstemplates is not documented.
/gpfs/dell2/emc/modeling/noscrub/George.Gayno/g2_lib.git/NCEPLIBS-g2-1/src/drstemplates.f:1: warning: Member maxtemp (variable) of class drstemplates is not documented.
```

These can be done in another PR.

**Comment by Hang-Lei-NOAA** [comment]
Yes,
The MAXOPTSSIZE warning cannot be addressed in this doxygen version.
I actually addressed drstemplates.f in another checkout, will commit.
Thanks,

**Comment by Hang-Lei-NOAA** [comment]
will spend sometime to see the The MAXOPTSSIZE warning can be addressed in this doxygen.

**Pull request merged**